python/pdo/contract/response.py

```diff
@@ -102,13 +103,17 @@ def commit_asynchronously(self, ledger_config=None, wait_parameter_for_ledger=30
         is automatically invoked to add a task for the second step
         """
 
+        # if the ledger is not set up then there is nothing to do here
+        if ledger_config == None :
+            return
+
         #start threads for commiting response if not done before
         if ContractResponse.__start_commit_service__:
             atexit.register(ContractResponse.__exit_commit_workers__)
 
             # start replication service
-            start_replication_service()
-            start_transaction_processing_service()
+            start_replication_service(ledger_config)
+            start_transaction_processing_service(ledger_config)
```

Comment:
not sure why ledger_config is need for starting replication service? I see that you use "config.get("replication_worker_threads", number_of_service_threads)" within start_replication_service, are you puttnig number of service threads within ledger config?

Reply:
i don't remember what it was used for per se... but you were referencing it previously. and... there needs to be a complete overhaul of the logic at some point so i dont think this is the big issue.